client/rpc-api/src/state/mod.rs

```diff
@@ -77,8 +77,6 @@ pub trait StateApi<Hash> {
 	fn child_storage_keys(
 		&self,
 		child_storage_key: StorageKey,
-		child_info: StorageKey,
```

Comment:
This is a breaking change to the RPC, we should probably plan some transition or add versioning, os that the JS libs can figure out which API version it's operating on.

Reply:
This is more related to this PR https://github.com/paritytech/substrate/pull/4857/files#r394314720 ,  there was a PR pending from @Stefie on the js side, I think the initial idea was that the api was never really well defined and do it in a breaking way, but the more time passes the more probable it will be needed to be versioned, I will look at how it can be versioned (I feel like as long as this makes it to 2.0 versioning is not needed but I may be wrong and #4857 may not make it to 2.0).

Reply:
Fair enough! We have a RPC call that returns a list of all available RPC methods (and their versions), but it's hardcoded to `1` for now. We don't really have a good mechanism to control that, but it's definitely possible.
If the API is not available on JS side yet (or not being used extensively) then I suppose it's probably fine to change without version change before 2.0), but if not then we need some migration path. I remember that JS team (notably @jacogr) had a really hard time maintaining different compatibility mechanisms.

Reply:
Ok, the issue is we cannot "just break" - the pending PR will not go in. We support different chains and cannot detect among them what params there are and what params there are not.

So as it stands, we will breaks some and not others. Sadly, not something that is quite acceptable.

Reply:
And by "fix", I mean the simplest is actually just a rename. No other magic required.

Reply:
Versioning will be better, I will have a look later. It is a tricky question to know how long deprecated interface would need support, maybe there is something to do by putting old api in some compatibility crate to avoid new project using them.
(Unlike runtime api there is no need to keep supporting old api forever)

Reply:
My PR is now a month old, now, so I also wouldn't just merge it anymore. I haven't touched it since I opened it.  I'll close it to avoid confusion.